core/reactor_common.c

```diff
@@ -1726,48 +1772,67 @@ int process_args(int argc, char* argv[]) {
                 usage(argc, argv);
                 return 0;
             }
-        } else if (strcmp(argv[i], "-k") == 0 || strcmp(argv[i], "--keepalive") == 0) {
-            if (argc < i + 2) {
+        } else if (strcmp(arg, "-k") == 0 || strcmp(arg, "--keepalive") == 0) {
+            if (argc < i + 1) {
                 error_print("--keepalive needs a boolean.");
                 usage(argc, argv);
                 return 0;
             }
-            i++;
-            char* keep_spec = argv[i];
+            char* keep_spec = argv[i++];
             if (strcmp(keep_spec, "true") == 0) {
                 keepalive_specified = true;
             } else if (strcmp(keep_spec, "false") == 0) {
                 keepalive_specified = false;
             } else {
                 error_print("Invalid value for --keepalive: %s", keep_spec);
             }
-        } else if (strcmp(argv[i], "-t") == 0 || strcmp(argv[i], "--threads") == 0) {
-            if (argc < i + 2) {
+        } else if (strcmp(arg, "-t") == 0 || strcmp(arg, "--threads") == 0) {
+            if (argc < i + 1) {
                 error_print("--threads needs an integer argument.s");
                 usage(argc, argv);
                 return 0;
             }
-            i++;
             char* threads_spec = argv[i++];
             int num_threads = atoi(threads_spec);
             if (num_threads <= 0) {
                 error_print("Invalid value for --threads: %s. Using 1.", threads_spec);
                 num_threads = 1;
             }
             _lf_number_of_threads = (unsigned int)num_threads;
-        } else if (strcmp(argv[i], "-i") == 0 || strcmp(argv[i], "--id") == 0) {
-            if (argc < i + 2) {
+        } else if (strcmp(arg, "-i") == 0 || strcmp(arg, "--id") == 0) {
+            if (argc < i + 1) {
                 error_print("--id needs a string argument.");
                 usage(argc, argv);
                 return 0;
             }
-            i++;
-            info_print("Federation ID for executable %s: %s", argv[0], argv[i]);
-            federation_id = argv[i++];
-        } else if (strcmp(argv[i], "--ros-args") == 0) {
+            federation_metadata.federation_id = argv[i++];
+            info_print("Federation ID for executable %s: %s", argv[0], federation_metadata.federation_id);
+        } else if (strcmp(arg, "--host") == 0) {
```

Comment:
I think an argument like `--rti` or `-r` might be more descriptive. How hard do you think it would be to parse a full address in C? For example `--rti user@host:port`?

Reply:
the primary reason I choose to separate --rti into --host --port and --user is to cut down the number of edge cases. Given we are parsing the address in C, doing --host --port and --user is a lot easier to get right.

Reply:
Right, I figured. Still, I think running a federate using a `--host` argument can be a bit misleading and hard to explain. Running a federate with a `--rti` argument I think is easier to understand and explain. It might be worth going the extra mile to parse a full host address.

I found a few promising ways to approach this problem just by googling around, including using [`sscanf`](https://stackoverflow.com/a/726214/783868) and/or using an [`isdigit()`](https://www.geeksforgeeks.org/isdigit-function-in-c-c-with-examples/) function that might be useful in `validate_port` as well.

If this is a dead-end, then I think `--rti --host xxx` is still a bit more descriptive that plain `--host`.